.github/workflows/ci-unit_tests.yaml

```diff
@@ -0,0 +1,60 @@
+name: CI Unit Tests
+on:
+    pull_request:
+
+jobs:
+
+  ci_pytest:
+      runs-on: ubuntu-latest
+      permissions:
+        checks: write
+        pull-requests: write
+
+      steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y perl libxml-libxml-perl libxml-libxslt-perl libdatetime-perl
+          python -m pip install --upgrade pip
+          pip install pytest wxflow
+          # If you have a requirements.txt file, you can install all dependencies using:
+          # pip install -r requirements.txt
+
+      - name: Cache Rocoto Install
+        uses: actions/cache@v4
+        with:
+          path: ~/rocoto
+          key: ${{ runner.os }}-rocoto-${{ hashFiles('**/ci-unit_tests.yaml') }}
+
+      - name: Install Rocoto
+        run: |
+          if [ ! -d "$HOME/rocoto/bin" ]; then
+            git clone https://github.com/christopherwharrop/rocoto.git $HOME/rocoto
+            cd $HOME/rocoto
+            ./INSTALL
+          fi
+          echo "$HOME/rocoto/bin" >> $GITHUB_PATH
+
+      - name: Run tests
+        run: |
+          cd $GITHUB_WORKSPACE/ci/scripts/tests
+          pytest -v --junitxml $GITHUB_WORKSPACE/ci/scripts/tests/test-results.xml
+
+      - name: List test directory
+        run: ls -la $GITHUB_WORKSPACE/ci/scripts/tests
+
+      - name: Publish Test Results
+        if: always()
+        uses: EnricoMi/publish-unit-test-result-action@v2
```

Comment:
It's up to you, I added it because I thought it was cool, what do you think? I'm not marred to it, we can take it out.  ([example](https://github.com/TerrenceMcGuinness-NOAA/global-workflow/actions/runs/9082547067))

Reply:
confirmed: `pull-requests: write` was needed to write out the GitHub Unit Test Report to the PR.

Reply:
I guess we'll leave it.